src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -50,233 +41,214 @@ export const useDiscoverHistogram = ({
-  const { fetchStatus: hitsFetchStatus, result: hitsTotal } = useDataState(
+  const firstLoadComplete = useRef(false);
+  const { query, filters, timeRange } = useQuerySubscriber({ data: services.data });
+  const { fetchStatus: totalHitsStatus, result: totalHitsResult } = useDataState(
     savedSearchData$.totalHits$
   );
 
-  const hits = useMemo(
-    () =>
-      isPlainRecord
-        ? undefined
-        : localHitsContext ?? {
-            status: hitsFetchStatus.toString() as UnifiedHistogramFetchStatus,
-            total: hitsTotal,
-          },
-    [hitsFetchStatus, hitsTotal, isPlainRecord, localHitsContext]
-  );
-
-  /**
-   * Chart
-   */
-
-  const onChartHiddenChange = useCallback(
-    (chartHidden: boolean) => {
-      storage.set(CHART_HIDDEN_KEY, chartHidden);
-      stateContainer.setAppState({ hideChart: chartHidden });
-    },
-    [stateContainer, storage]
-  );
-
-  const onChartLoad = useCallback(
-    (event: UnifiedHistogramChartLoadEvent) => {
-      // We need to store the Lens request adapter in order to inspect its requests
-      inspectorAdapters.lensRequests = event.adapters.requests;
-    },
-    [inspectorAdapters]
-  );
-
-  const chart = useMemo(
-    () =>
-      isPlainRecord || !isTimeBased
-        ? undefined
-        : {
-            hidden: hideChart,
-            timeInterval: interval,
-          },
-    [hideChart, interval, isPlainRecord, isTimeBased]
-  );
-
-  // Clear the Lens request adapter when the chart is hidden
   useEffect(() => {
-    if (hideChart || !chart) {
-      inspectorAdapters.lensRequests = undefined;
+    if (!unifiedHistogram?.initialized) {
+      return;
     }
-  }, [chart, hideChart, inspectorAdapters]);
-
-  /**
-   * Breakdown
-   */
 
-  const onBreakdownFieldChange = useCallback(
-    (newBreakdownField: DataViewField | undefined) => {
-      stateContainer.setAppState({ breakdownField: newBreakdownField?.name });
-    },
-    [stateContainer]
-  );
+    let newState: Partial<UnifiedHistogramState> = {
+      dataView,
+      query,
+      filters,
+      timeRange,
+      searchSessionId,
+      requestAdapter: inspectorAdapters.requests,
+    };
 
-  const field = useMemo(
-    () => (breakdownField ? dataView.getFieldByName(breakdownField) : undefined),
-    [dataView, breakdownField]
-  );
+    // We only want to show the partial results on the first load,
+    // or there will be a flickering effect as the loading spinner
+    // is quickly shown and hidden again on fetches
+    if (!firstLoadComplete.current) {
+      newState = {
+        ...newState,
+        totalHitsStatus: totalHitsStatus.toString() as UnifiedHistogramFetchStatus,
+        totalHitsResult,
+      };
+    }
 
-  const breakdown = useMemo(
-    () => (isPlainRecord || !isTimeBased ? undefined : { field }),
-    [field, isPlainRecord, isTimeBased]
-  );
+    unifiedHistogram.updateState(newState);
```

Comment:
Ideally we make this a method call off the state service

Reply:
I'd like to avoid exposing the state service to the consumer directly in favour of working with the API object that includes select state service functionality, but I've simplified how these methods are exposed by simply picking the desired functions from the state service and spreading them over the API object.